Subject: [PATCH] primeiro projeto
---
Index: src/main/java/minimundo/com/Funcionário.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/minimundo/com/Funcionário.java b/src/main/java/minimundo/com/Funcionário.java
new file mode 100644
--- /dev/null	(date 1743983480023)
+++ b/src/main/java/minimundo/com/Funcionário.java	(date 1743983480023)
@@ -0,0 +1,56 @@
+package minimundo.com;
+
+import java.util.Scanner;
+
+public class Funcionário {
+    // atributos
+    String nome;
+    float salario;
+    private String cpf;
+
+   // construtores
+    public Funcionário(){
+        nome="";
+        salario=0;
+         cpf="";
+    }
+    public Funcionário(String nome, float salario, String cpf){
+        this.nome= nome;
+        this.salario=salario;
+        this.cpf=cpf;
+    }
+    //metodos gets e set
+    public String getNome(){
+        return nome;
+    }
+    public void setNome(String nome){
+        this.nome=nome;
+    }
+    public float getSalario(){
+        return salario;
+    }
+    public void setSalario(float salario){
+        this.salario=salario;
+    }
+    public String getCpf(){
+        return cpf;
+    }
+    public void setCpf(String cpf ){
+        this.cpf=cpf;
+    }
+    public void perguntar(){
+        Scanner perguntar= new Scanner(System.in);
+        System.out.print("Qual é seu nome: ");
+        nome= perguntar.next();
+        System.out.print("Qual é seu salario ");
+        salario= perguntar.nextInt();
+        System.out.print("Digite seu cpf: ");
+        cpf= perguntar.next();
+    }
+    // metodo da lógica do negocio
+    public  void apresentar(){
+        System.out.println("Nome: "+nome);
+        System.out.println("Salário: "+salario);
+        System.out.println("CPF: "+cpf);
+    }
+}
Index: .gitignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.gitignore b/.gitignore
new file mode 100644
--- /dev/null	(date 1743289327844)
+++ b/.gitignore	(date 1743289327844)
@@ -0,0 +1,38 @@
+target/
+!.mvn/wrapper/maven-wrapper.jar
+!**/src/main/**/target/
+!**/src/test/**/target/
+
+### IntelliJ IDEA ###
+.idea/modules.xml
+.idea/jarRepositories.xml
+.idea/compiler.xml
+.idea/libraries/
+*.iws
+*.iml
+*.ipr
+
+### Eclipse ###
+.apt_generated
+.classpath
+.factorypath
+.project
+.settings
+.springBeans
+.sts4-cache
+
+### NetBeans ###
+/nbproject/private/
+/nbbuild/
+/dist/
+/nbdist/
+/.nb-gradle/
+build/
+!**/src/main/**/build/
+!**/src/test/**/build/
+
+### VS Code ###
+.vscode/
+
+### Mac OS ###
+.DS_Store
\ No newline at end of file
Index: .idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/misc.xml b/.idea/misc.xml
new file mode 100644
--- /dev/null	(date 1743289331593)
+++ b/.idea/misc.xml	(date 1743289331593)
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ExternalStorageConfigurationManager" enabled="true" />
+  <component name="MavenProjectsManager">
+    <option name="originalFiles">
+      <list>
+        <option value="$PROJECT_DIR$/pom.xml" />
+      </list>
+    </option>
+  </component>
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_21" default="true" project-jdk-name="21" project-jdk-type="JavaSDK">
+    <output url="file://$PROJECT_DIR$/out" />
+  </component>
+</project>
\ No newline at end of file
Index: src/main/java/minimundo/com/Desenvolvedor.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/minimundo/com/Desenvolvedor.java b/src/main/java/minimundo/com/Desenvolvedor.java
new file mode 100644
--- /dev/null	(date 1743983822278)
+++ b/src/main/java/minimundo/com/Desenvolvedor.java	(date 1743983822278)
@@ -0,0 +1,64 @@
+package minimundo.com;
+
+import java.util.Scanner;
+
+public class Desenvolvedor extends Funcionário {
+    // atributos
+    private String linguagem;
+    private String framework;
+
+    public Desenvolvedor() {
+        linguagem = "";
+        framework = "";
+    }
+    public Desenvolvedor(String nome, float salario, String cpf, String linguagem, String framework) {
+       super(nome, salario, cpf);
+       this.linguagem=linguagem;
+       this.framework=framework;
+    }
+
+    public String getLinguagem() {
+        return linguagem;
+    }
+
+    public void setLinguagem(String linguagem) {
+        this.linguagem = linguagem;
+    }
+
+    public String getFramework() {
+        return framework;
+    }
+
+    public void setFramework(String framework) {
+        this.framework = framework;
+    }
+    @Override
+public void perguntar (){
+        Scanner perguntar= new Scanner(System.in);
+      super.perguntar();
+        System.out.print("Qual linguagem você programa: ");
+        linguagem=perguntar.nextLine();
+        System.out.print("Qual framework você utiliza? ");
+        framework=perguntar.nextLine();
+}
+    // polimorfismo- sobreescritak
+    @Override
+    public void apresentar() {
+        super.apresentar();
+        System.out.println("Linguagem: " + linguagem);
+        System.out.println("Framework: " + framework);
+    }
+
+    //polimorfismo- sobrecarga(overload)
+    public void programar() {
+        System.out.println("Estounprogramando.....");
+    }
+
+    public void programar(String opcao) {
+        System.out.println("Estounprogramando mais rapido");
+    }
+
+    public void programar(int opcao) {
+        System.out.println("Estounprogramando em" + opcao + "x");
+    }
+}
Index: .idea/.gitignore
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/.gitignore b/.idea/.gitignore
new file mode 100644
--- /dev/null	(date 1743289335795)
+++ b/.idea/.gitignore	(date 1743289335795)
@@ -0,0 +1,3 @@
+# Default ignored files
+/shelf/
+/workspace.xml
Index: src/main/java/minimundo/com/Main.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/minimundo/com/Main.java b/src/main/java/minimundo/com/Main.java
new file mode 100644
--- /dev/null	(date 1743984602140)
+++ b/src/main/java/minimundo/com/Main.java	(date 1743984602140)
@@ -0,0 +1,9 @@
+package minimundo.com;
+
+public class Main {
+    public static void main(String[] args) {
+        Desenvolvedor dev=new Desenvolvedor();
+       dev.perguntar();
+       dev.apresentar();
+        }
+    }
Index: pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pom.xml b/pom.xml
new file mode 100644
--- /dev/null	(date 1743289341220)
+++ b/pom.xml	(date 1743289341220)
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>dobackaofront</groupId>
+    <artifactId>Objetos</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <properties>
+        <maven.compiler.source>21</maven.compiler.source>
+        <maven.compiler.target>21</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+</project>
\ No newline at end of file
Index: .idea/vcs.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.idea/vcs.xml b/.idea/vcs.xml
new file mode 100644
--- /dev/null	(date 1743986435980)
+++ b/.idea/vcs.xml	(date 1743986435980)
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+    <mapping directory="$PROJECT_DIR$/POO" vcs="Git" />
+  </component>
+</project>
\ No newline at end of file
Index: src/main/java/minimundo/com/Empresa.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/minimundo/com/Empresa.java b/src/main/java/minimundo/com/Empresa.java
new file mode 100644
--- /dev/null	(date 1743969693140)
+++ b/src/main/java/minimundo/com/Empresa.java	(date 1743969693140)
@@ -0,0 +1,5 @@
+package minimundo.com;
+
+public class Empresa {
+    // tem varios departamentos
+}
Index: src/main/java/minimundo/com/Departamento.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/minimundo/com/Departamento.java b/src/main/java/minimundo/com/Departamento.java
new file mode 100644
--- /dev/null	(date 1743969597258)
+++ b/src/main/java/minimundo/com/Departamento.java	(date 1743969597258)
@@ -0,0 +1,4 @@
+package minimundo.com;
+
+public class Departamento {
+}
